libmamba/include/mamba/core/subdirdata.hpp

```diff
@@ -32,6 +32,19 @@ namespace decompress
 namespace mamba
 {
 
+    // Represents a repodata update "jlap" document
+    struct Jlap
+    {
+        std::string start_hash;
+        std::string end_hash;
+        nlohmann::json metadata;
+        std::vector<nlohmann::json> patches;
+
+        static tl::expected<Jlap, std::runtime_error> parse(const fs::u8path& path);
```

Comment:
I think the error type should be `mamba_error` and depending on which `mamba_error_code` are available, maybe another one need to be added, not sure.